### ISSUE TITLE
Using a Colorbar for categorical data OR forcing grouping for continuous data

### ISSUE THREAD

**Issue by Krastanov** [MakieOrg/AlgebraOfGraphics.jl#282]
Consider a data frame with three columns: x value, y value, color.

```julia
using DataFrames, AlgebraOfGraphics, CairoMakie
x = [1,2,3,4]
y0 = [1,2,3,4]
y1 = y0.+1
df = DataFrame(x=[x;x],y=[y0;y1],color=[0,0,0,0,1,1,1,1])
```

I want to make a line plot in which we have **grouping into separate lines** like this:

```julia
data(df) * mapping(:x,:y) * visual(Lines) * mapping(color=:color => nonnumeric) |> draw
```
![image](https://user-images.githubusercontent.com/705248/134789462-7168e7b8-fb24-45e8-9517-74a901a4e5fb.png)

But because I have many different colors I want a **continuous color scale** (which here regrettably does not provide separate lines):

```julia
data(df) * mapping(:x,:y) * visual(Lines) * mapping(color=:color) |> draw
```
![image](https://user-images.githubusercontent.com/705248/134789482-6dbf420d-d62e-4397-acbd-3c9b09d69baf.png)

Basically, each line is guaranteed to have the same color (so categorical treatment of the color makes sense), but there are so many categories and the categories have a notion of distance between them, that they should be shown on a colorbar, not a legend.

Here is a popular type of a plot like this:
https://user-images.githubusercontent.com/705248/134789512-2ad93866-013c-4ad6-ab32-5bb1a90af12f.png

So my question is: **How do I force a continuous color palette and a colorbar for a categorical color mapping? Or equivalently, how do I force a grouping for a continuous color mapping?**

**Comment by greimel** [comment]
You can do

```julia
data(df) * mapping(:x,:y) * visual(Lines) * mapping(color=:col, group=:col => nonnumeric) |> draw
```

<img width="418" alt="image" src="https://user-images.githubusercontent.com/6280307/137643042-ba5ad062-19ce-43cc-ac76-ce14663bb07f.png">

**Comment by greimel** [comment]
xref #183 for categorical colorbar

**Comment by jkrumbiegel** [comment]
This is now explicitly mentioned in the docs
